packages/effect/src/Encoding.ts

```diff
@@ -88,6 +88,23 @@ export const decodeHex = (str: string): Either.Either<Uint8Array, DecodeExceptio
  */
 export const decodeHexString = (str: string) => Either.map(decodeHex(str), (_) => Common.decoder.decode(_))
 
+/**
+ * Encodes a UTF-8 `string` into a URI component `string`.
+ *
+ * @category encoding
+ * @since 3.11.0
+ */
+export const encodeUriComponent = (str: string) => encodeURIComponent(str)
```

Comment:
it must be wrapped as it throws [developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#exceptions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#exceptions)

Reply:
Thanks for the feedbacks, I have updated my code accordingly